src/Sulu/Bundle/MediaBundle/Media/ImageConverter/ImagineImageConverter.php

```diff
@@ -12,116 +12,172 @@
+            return new ImagickImagine();
+        } catch (RuntimeException $ex) {
+            return new GdImagine();
+        }
+    }
+
+    /**
+     * Applies an array of transformations on a passed image.
+     *
+     * @param ImageInterface $image
+     * @param $tansformations
+     *
+     * @throws ImageProxyInvalidFormatOptionsException
+     *
+     * @return ImageInterface The modified image
+     */
+    private function applyTransformations(ImageInterface $image, $tansformations)
+    {
+        foreach ($tansformations as $transformation) {
+            if (!isset($transformation['effect'])) {
+                throw new ImageProxyInvalidFormatOptionsException('Effect not found');
             }
+            $image = $this->call($image, $transformation['effect'], $transformation['parameters']);
         }
 
-        return $this->image;
+        return $image;
+    }
+
+    /**
+     * Scales a given image according to the information passed as the second argument.
+     *
+     * @param ImageInterface $image
+     * @param $scale
+     *
+     * @return ImageInterface
+     */
+    private function applyScale(ImageInterface $image, $scale)
+    {
+        $parameters = [
+            'x' => $scale['x'],
+            'y' => $scale['y'],
+            'mode' => $scale['mode'],
+        ];
+
+        return $this->scaleTransformation->execute($image, $parameters);
     }
 
     /**
-     * set the image palette to RGB.
+     * Ensures that the color mode of the passed image is RGB.
+     *
+     * @param ImageInterface $image
+     *
+     * @return ImageInterface $image The modified image
      */
-    protected function toRGB()
+    private function toRGB(ImageInterface $image)
     {
-        if ($this->image->palette()->name() == 'cmyk') {
-            $this->image->usePalette(new RGB());
+        if ($image->palette()->name() == 'cmyk') {
+            $image->usePalette(new RGB());
         }
+
+        return $image;
     }
 
     /**
-     * @param $command
+     * Calls a given transformation with given parameters on the passed image.
+     *
+     * @param ImageInterface $image
+     * @param $transformation
      * @param $parameters
      *
-     * @throws ImageProxyInvalidFormatOptionsException
+     * @return ImageInterface $image The modified image
      */
-    public function call($command, $parameters)
+    private function call(ImageInterface $image, $transformation, $parameters)
     {
-        if (count($this->image->layers())) {
-            $counter = 0;
-            $this->image->layers()->coalesce();
-            foreach ($this->image->layers() as $layer) {
-                ++$counter;
-                $this->commandManager->get($command)->execute($layer, $parameters);
-                if ($counter == 1) {
-                    /** @var \Imagine\Imagick\Image|\Imagine\Gd\Image $image */
-                    $image = $layer; // use first layer as main image
+        if (count($image->layers())) {
```

Comment:
@alexander-schranz Wasn't that implemented in the `LiipImagine` library we are using? So is this code useless?